PyPDF2/_merger.py

```diff
@@ -276,9 +287,6 @@ def write(self, fileobj: StrByteType) -> None:
         # Write the output to the file
         self.output.write(fileobj)
 
-        if my_file:
-            fileobj.close()
-
```

Comment:
Removing this might break existing workflows. Please add it back. 

The hard thing about developing for a library that is already in heavy use is that we need to be really careful not to break anything.

We can make breaking changes, but we need to announce it + have a phase in which we support both ways: https://pypdf2.readthedocs.io/en/latest/dev/deprecations.html